src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -4203,11 +4203,26 @@ void CmdSketcherConstrainTangent::activated(int iMsg)
 
         if (isVertex(GeoId1,PosId1) && isVertex(GeoId2,PosId2)) { // endpoint-to-endpoint tangency
 
+            if (isBsplineKnot(Obj, GeoId2)) {
```

Comment:
@abdullahtahiriyo while I have added a logic to prevent it in GUI, it is very much possible to pass the points inverted through python (so that the line end point is first). This is also possible in other cases. How do you want to proceed with it? The logic can start getting pretty convoluted.

Reply:
The current approach is:
1. A UI user should be guided and prevented from getting hurt.
2. A Python power user knows FC enough. Bad output as a result of bad input is intended behaviour.

Any alternative to this approach I think is very convoluted.